src/modules/cmdpal/Microsoft.CmdPal.UI/ExtViews/ListPage.xaml.cs

```diff
@@ -160,20 +167,32 @@ public void Receive(ActivateSelectedListItemMessage message)
         }
     }
 
-    private void ItemsList_SelectionChanged(object sender, SelectionChangedEventArgs e)
+    private static void OnViewModelChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
     {
-        if (ItemsList.SelectedItem is ListItemViewModel item)
+        if (d is ListPage @this)
         {
-            ViewModel?.UpdateSelectedItemCommand.Execute(item);
+            if (e.OldValue is ListViewModel old)
+            {
+                old.PropertyChanged -= @this.ViewModel_PropertyChanged;
+            }
+
+            if (e.NewValue is ListViewModel page)
+            {
+                page.PropertyChanged += @this.ViewModel_PropertyChanged;
+            }
         }
     }
 
-    public void Receive(ShowExceptionMessage message)
+    private void ViewModel_PropertyChanged(object? sender, System.ComponentModel.PropertyChangedEventArgs e)
     {
-        _ = _queue.EnqueueAsync(() =>
+        var prop = e.PropertyName;
+        if (prop == nameof(ViewModel.ErrorMessage) && ViewModel != null)
         {
-            LoadedState = ViewModelLoadedState.Error;
-        });
+            if (!string.IsNullOrEmpty(ViewModel.ErrorMessage))
+            {
+                LoadedState = ViewModelLoadedState.Error;
+            }
+        }
```

Comment:
Future note: I should have looked at what this is doing. If all we use `LoadedState` for us the error display now; we can just pivot the UI based on the ErrorMessage being not blank itself in XAML and avoid the event chaining here.